File: src/main/java/Ex13_Map.java
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

import static java.util.Collections.singletonMap;

public class Ex13_Map {
    public static void main(String[] args) {
        Map<String, Integer> singleFruitWithSugar = singletonMap("Kiwi", 60); // Singleton Map of fruit => Immutable

        Map<String, Integer> fruitWithSugar  = new HashMap<>() {{ // Mutable Map of fruits associated by sugar
            put("Apple", 50);
            put("Banana", 70);
        }};

        Map<String, Integer> otherFruitWithSugar = Map.of( // Immutable Map of fruits associated by sugar, limited to 10 Entry
                "Avocado", 30,
                "Ananas", 70,
                "Cranberry", 50
        );

        Map<Object, Object> noFruitWithSugar = Collections.emptyMap(); // Empty Map of fruits associated by sugar
    }
}

File: src/main/kotlin/Ex13_Map.kt
fun main() {
  val singleFruitWithSugarImmutable: Map<String, Int> = mapOf("Kiwi" to 60) // Immutable singleton Map of fruit
  val singleFruitWithSugarMutable: MutableMap<String, Int> = mutableMapOf("Kiwi" to 60) // Mutable singleton Map of fruit

  val fruitWithSugar: MutableMap<String, Int> = mutableMapOf( // Mutable Map of fruits associated by sugar
    "Apple" to 50,
    "Banana" to 70
  )

  val otherFruitWithSugar: Map<String, Int> = mapOf( // Immutable Map of fruits associated by sugar, no limit here like on Java
    "Avocado" to 30,
    "Ananas" to 70,
    "Cranberry" to 50
  )

  val noFruitWithSugarI: Map<String, Int> = mapOf() // Immutable Empty Map of fruits associated by sugar
  val noFruitWithSugarM: Map<String, Int> = mutableMapOf() // Mutable Empty Map of fruits associated by sugar
}